core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreMainActivity.java

```diff
@@ -1081,18 +1090,38 @@ private void openAndSetInContainer(File file) {
       e.printStackTrace();
     }
     zimReaderContainer.setZimFile(file);
-    if (zimReaderContainer.getZimFileReader() != null) {
+    final ZimFileReader zimFileReader = zimReaderContainer.getZimFileReader();
+    if (zimFileReader != null) {
       if (mainMenu != null) {
-        mainMenu.onFileOpened(zimReaderContainer.getZimFileReader(), !urlIsInvalid());
+        mainMenu.onFileOpened(zimFileReader, !urlIsInvalid());
       }
       openMainPage();
-      presenter.loadCurrentZimBookmarksUrl();
+      safeDispose();
+      bookmarkingDisposable = Flowable.combineLatest(
+        newBookmarksDao.bookmarkUrlsForCurrentBook(zimFileReader),
+        webUrlsProcessor,
+        (bookmarkUrls, currentUrl) -> bookmarkUrls.contains(currentUrl)
+      ).observeOn(AndroidSchedulers.mainThread())
+        .subscribe(isBookmarked -> {
+            this.isBookmarked = isBookmarked;
+            bottomToolbarBookmark.setImageResource(
+              isBookmarked ? R.drawable.ic_bookmark_24dp : R.drawable.ic_bookmark_border_24dp);
+          },
+          Throwable::printStackTrace
```

Comment:
Is there a difference between this & Log.d? Are there any best practices around it?

Reply:
It prints the stacktrace to console. In a previous project we had an error function written and it was something like
```
if (debug) throw Exception
else printStackTrace
```
It depends very much on your use case, ask yourself the question "should my application crash, retry, report an error or silently fail"

The difference between this and 3 param Log.x is the 1st/2nd parameter which allows you to add some context about where this crash happened ie tag and message. I am using this because the method reference is quicker to write. Best practice would probably be more verbose logging or indeed to simply crash because something must have gone wrong